Minor fixes to build and packaging

martinsumner opened the pull request:
See https://github.com/nhs-riak/riak/issues/1

aef- committed:
echo is not the most portable method to pipe output. In some systems, `echo -n hello` will output `"-n hello"`. `printf` is a more reliable way to "echo" variables without newlines.

martinsumner committed:
Minor fixes to build and packaging

See https://github.com/nhs-riak/riak/issues/1

stephenwood24-nhs commented on Makefile:
```diff
@@ -242,6 +242,7 @@ PKG_ID := "$(REPO_TAG)-OTP$(OTP_VER)"
 PKG_VERSION = $(shell echo $(PKG_ID) | sed -e 's/^$(REPO)-//')
 
 package:
+	compile
```
I don't think this `compile` is neccessary. I've built debian package dozens of times with `make package` and had no issues

martinsumner committed:
Place compile on correct line

martinsumner committed:
revert no-lintian

Errors should not generally prevent the production of packages

martinsumner committed:
Merge remote-tracking branch 'aef-/aef-/update-riak-admin-escape' into nhse-d32-nhs.i1

martinsumner commented:
The change to the break command doesn't work as expected, in that we're changing the type of a previous configuration parameter - so anyone copying over a previous riak.conf will get a `Cuttlefish failed!` message if they attempt to start

martinsumner committed:
Change break handler to translation

This means it should work with existing configurations setting the ignore_break_single to true/false

tburghart commented on priv/riak.schema:
```diff
@@ -118,11 +118,23 @@
+    Setting = cuttlefish:conf_get("erlang.vm.ignore_break_signal", Conf),
+    case Setting of
+      true -> "i";
+      false -> "c"
```
I think `false` should call `cuttlefish:unset()`

martinsumner replied:
Changed as suggested, looking at `ps` output after starting riak:

Setting to `true` -

```riak -scl false -sfwi 500 -P 256000 -e 256000 -Q 262144 -A 64 -K true -W w -B i -zdbbl 32768 ...```

Setting to `false` -

```riak -scl false -sfwi 500 -P 256000 -e 256000 -Q 262144 -A 64 -K true -W w -zdbbl 32768 ...```

martinsumner committed:
Unset if break_signal false
